src/main/kotlin/no/nav/familie/ba/sak/kjerne/behandling/behandlingstema/BehandlingstemaUtils.kt

```diff
@@ -10,14 +13,44 @@ import no.nav.familie.ba.sak.kjerne.tidslinje.komposisjon.innholdForTidspunkt
 import no.nav.familie.ba.sak.kjerne.tidslinje.tid.MånedTidspunkt
 import no.nav.familie.ba.sak.kjerne.vilkårsvurdering.domene.Regelverk
 
+fun bestemKategoriVedOpprettelse(
+    overstyrtKategori: BehandlingKategori?,
+    behandlingType: BehandlingType,
+    behandlingÅrsak: BehandlingÅrsak,
+    // siste iverksatt behandling som har løpende utbetaling. Hvis løpende utbetaling ikke finnes, settes det til NASJONAL
+    kategoriFraLøpendeBehandling: BehandlingKategori
+): BehandlingKategori {
+    val behandlingsTypeNavn = behandlingType.visningsnavn
+    return when {
+        behandlingType == BehandlingType.FØRSTEGANGSBEHANDLING ||
+            behandlingType == BehandlingType.REVURDERING && behandlingÅrsak == BehandlingÅrsak.SØKNAD -> {
+            overstyrtKategori
+                ?: throw FunksjonellFeil(
+                    "Behandling med type $behandlingsTypeNavn krever behandlingskategori",
+                )
+        }
+        behandlingType == BehandlingType.MIGRERING_FRA_INFOTRYGD && behandlingÅrsak.erFørstegangMigreringsårsak() -> {
+            BehandlingKategori.NASJONAL
+        }
+        else -> {
+            kategoriFraLøpendeBehandling
+        }
```

Comment:
Kunne vi gjort det eksplisitt hva else er her? Så blir man tvunget til å ta stilling til dette om vi får en ny behandlingskategori

```suggestion
        MIGRERING_FRA_INFOTRYGD_OPPHØRT,
        TEKNISK_OPPHØR,
        TEKNISK_ENDRING -> {
            kategoriFraLøpendeBehandling
        }
```